src/code.jl

```diff
@@ -146,11 +146,13 @@ function function_to_expr(op::typeof(^), O, st)
             return toexpr(Term(inv, Any[ex]), st)
         else
             args = Any[Term(inv, Any[ex]), -args[2]]
-            op = get(st.rewrites, :nanmath, false) ? op : NaNMath.pow
+            op = get(st.rewrites, :nanmath, false) || args[2] isa Integer ? op : NaNMath.pow
```

Comment:
Do we need this at all, see https://github.com/JuliaSymbolics/SymbolicUtils.jl/pull/689? In any case, the existing check is wrong and should be
```suggestion
            op = get(st.rewrites, :nanmath, false) === true && !(args[2] isa Integer) ? NaNMath.pow : op
```